service/controller/app/v1/resource/namespace/desired.go

```diff
@@ -0,0 +1,55 @@
+package namespace
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/giantswarm/microerror"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/giantswarm/app-operator/pkg/label"
+	"github.com/giantswarm/app-operator/pkg/project"
+	"github.com/giantswarm/app-operator/service/controller/app/v1/key"
+)
+
+func (r *Resource) GetDesiredState(ctx context.Context, obj interface{}) (interface{}, error) {
+	cr, err := key.ToCustomResource(obj)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	var clusterID, org string
+	{
+		name := key.ClusterValuesConfigMapName(cr)
+		cm, err := r.k8sClient.CoreV1().ConfigMaps(cr.Namespace).Get(name, metav1.GetOptions{})
```

Comment:
Fetching `clusterconfigmap` for getting labels from it.

Reply:
Why does the `App` CR not just have the cluster ID and organization ID labels?

Reply:
@xh3b4sd `app` CRs are located in control plane, so it supposed to not contain `cluster` label, although I'm not sure `organization` at there. 
We could fetch two labels on `clusterconfigmap` so I chose to fetching from this object.

Reply:
I don't follow that. Check any resource in the CP we manage for the TCs. They all have these labels.

Reply:
Also see https://github.com/giantswarm/fmt/blob/master/kubernetes/annotations_and_labels.md#common-labels.

Reply:
@xh3b4sd You are right
```
 k get apps --all-namespaces -o yaml
apiVersion: v1
items:
- apiVersion: application.giantswarm.io/v1alpha1
  kind: App
  metadata:
    annotations:
      chart-operator.giantswarm.io/force-helm-upgrade: "true"
    creationTimestamp: "2019-08-29T07:50:00Z"
    finalizers:
    - operatorkit.giantswarm.io/app-operator
    generation: 1
    labels:
      app: kube-state-metrics
      app-operator.giantswarm.io/version: 1.0.0
      giantswarm.io/cluster: mbnc3
      giantswarm.io/managed-by: cluster-operator
      giantswarm.io/organization: giantswarm
      giantswarm.io/service-type: managed
    name: kube-state-metrics
```
I can make it to work with `app` CR, not involved any configMaps in here, Thanks.

Reply:
@rossf7 while analyzing some alerts and related logs, I found examples of Apps where assumption above about common labels being present on the App CRs does not hold - on anubis CP there are currently 3 Apps (2 aqua-app-enforcer Apps for 2 TCs, and 1 cluster-operator-2.0.0 App as configured in [kvm-app-collection](https://github.com/giantswarm/kvm-app-collection/blob/master/helm/kvm-app-collection-chart/templates/cluster-operator-2.0.0.yaml#L6)), all of which have only `app-operator.giantswarm.io/version=1.0.0` label; because of this log messages in e.g. in `namespace/current.go` logging cluster ID currently log empty strings there. On the other hand two`*-cluster-values` ConfigMaps have both `giantswarm.io/cluster` and `giantswarm.io/organization` label set.

Are these Apps without standard labels temporary exception? If not, should we switch back to deriving cluster and organization info from cluster values ConfigMaps?

Reply:
Yes these app CRs are being created by different components which is why the labels are not consistent.

Having gatekeeper do the defaulting would help here. @puja108 WDYT?

aqua-app-enforcer was probably created by the api and it misses adding the cluster label.
https://github.com/giantswarm/api/blob/260f363ae1d94d088aabb487a0a96d71362df050/service/app/creator/service.go#L102

The cluster-operator CR is probably because the helm chart misses the label. The namespace resource only interacts with tenant clusters but we need to make sure the resource doesn't error and logs OK.

Reply:
The cluster values CM is not used by VOO app CRs. So I think we should add the missing labels.